Shared/Samples/Show viewshed from geoelement in scene/ShowViewshedFromGeoelementInSceneView.swift

```diff
@@ -0,0 +1,200 @@
+        /// The graphic for the tank.
+        let tankGraphic: Graphic = {
+            let tankSymbol = ModelSceneSymbol(url: .bradleyTank, scale: 10)
+            tankSymbol.heading = 90
+            tankSymbol.anchorPosition = .bottom
+            let tankGraphic = Graphic(
+                geometry: Point(x: -4.506390, y: 48.385624, spatialReference: .wgs84),
+                attributes: ["HEADING": 0.0],
+                symbol: tankSymbol
+            )
+            return tankGraphic
+        }()
+        
+        /// The point for the tank to move toward.
+        var waypoint: Point!
+        
+        init() {
+            // Create camera controller.
+            cameraController = OrbitGeoElementCameraController(
+                target: tankGraphic,
+                distance: 200
+            )
+            
+            // Add tank graphic to graphics overlay.
+            graphicsOverlay.addGraphic(tankGraphic)
+            
+            // Create a viewshed to attach to the tank.
+            let geoElementViewshed = GeoElementViewshed(
+                geoElement: tankGraphic,
+                horizontalAngle: 90,
+                verticalAngle: 40,
+                headingOffset: 0,
+                pitchOffset: 0,
+                minDistance: 0.1,
+                maxDistance: 250
+            )
+            
+            // Offset viewshed observer location to top of tank.
+            geoElementViewshed.offsetZ = 3
+            
+            // Add the viewshed to the analysisOverlay to add to the scene.
+            analysisOverlay.addAnalysis(geoElementViewshed)
+        }
+        
+        /// Animate the tank moving from its current point to the waypoint.
+        func animate() {
+            // Get point from the current tank position.
+            guard let tankLocation = tankGraphic.geometry as? Point else { return }
+            
+            // Get the distance from the current tank location to the waypoint.
+            guard let distanceResult = GeometryEngine.geodeticDistance(
+                from: tankLocation,
+                to: waypoint,
+                distanceUnit: .meters,
+                azimuthUnit: .degrees,
+                curveType: .geodesic
+            ) else { return }
+            
+            // Move toward waypoint a short distance.
+            let locations = GeometryEngine.geodeticMove(
+                [tankLocation],
+                distance: 1,
+                distanceUnit: .meters,
+                azimuth: distanceResult.azimuth1.value,
+                azimuthUnit: distanceResult.azimuth1.unit.angularUnit,
+                curveType: .geodesic
+            )
+            tankGraphic.geometry = locations.first
+            
+            // Set tank graphic heading.
+            if let heading = tankGraphic.attributes["HEADING"] as? Double {
+                tankGraphic.setAttributeValue(
+                    heading + ((distanceResult.azimuth1.value - heading) / 10),
```

Comment:
I think it is worth adding a comment noting that "divided by 10" is merely for making the heading adjustment animation 10 times more smooth, instead of just jumping directly to the correct heading. (took me a few minutes to figure out 😅)